bfvmm/src/vmcs/src/vmcs_intel_x64_check_controls.cpp

```diff
@@ -379,17 +409,27 @@ vmcs_intel_x64::check_control_enable_pml_checks()
 void
 vmcs_intel_x64::check_control_unrestricted_guests()
 {
-    if (!is_enabled_unrestricted_guests())
+    using namespace vmcs::primary_processor_based_vm_execution_controls;
+    using namespace vmcs::secondary_processor_based_vm_execution_controls;
+
+    bool secondary_ctls = activate_secondary_controls::is_enabled();
+
+    if (!unrestricted_guest::is_enabled() || !secondary_ctls)
         return;
 
-    if (!is_enabled_ept())
+    if (!enable_ept::is_enabled())
         throw std::logic_error("enable ept must be 1 if unrestricted guest is 1");
 }
 
 void
 vmcs_intel_x64::check_control_enable_vm_functions()
 {
-    if (!is_enabled_vm_functions())
+    using namespace vmcs::primary_processor_based_vm_execution_controls;
+    using namespace vmcs::secondary_processor_based_vm_execution_controls;
+
+    bool secondary_ctls = activate_secondary_controls::is_enabled();
+
+    if (!enable_vm_functions::is_enabled() || !secondary_ctls)
```

Comment:
same thing here... two different tests, no `bool` and should be using `if_exists()` versions for all of the secondary ones. 

The reality is... your going to have to do through these checks and make sure that you clean the above up as I bet there are several spots where this is messed up. And likely... you can just use the `if_exists` for most of these to keep it clean.